spec/jobs/searchgov_domain_destroyer_job_spec.rb

```diff
@@ -1,38 +1,90 @@
 require 'spec_helper'
 
 describe SearchgovDomainDestroyerJob do
-  subject(:perform) { described_class.perform_now(**args) }
-
-  let(:domain) { 'www.archive.gov' }
-  let!(:searchgov_domain) {
-    SearchgovDomain.create!(domain: domain, status: 200)
-  }
-  let(:args) {
-    { searchgov_domain: searchgov_domain }
-  }
-
-  describe '#perform' do
-    it 'requires a searchgov_domain as an argument' do
-      expect{ described_class.perform_now }.
-        to raise_error(ArgumentError, /missing keyword: :?searchgov_domain/)
+  subject(:perform) { described_class.perform_now(searchgov_domain) }
+
+  let(:searchgov_domain) { SearchgovDomain.create(domain: "www.archive.gov", status: "200 OK") }
+  let(:error_messages) { [] }
+
+  before do
+    allow(Rails.logger).to receive(:error) { |message| error_messages << message }
+  end
+
+  describe "#perform" do
+    it "requires a searchgov_domain as an argument" do
+      expect { described_class.perform_now }.to raise_error(ArgumentError)
     end
 
-    it 'destroys the searchgov domain' do
-      expect(searchgov_domain).to receive(:destroy!)
+    it "destroys the searchgov domain" do
       perform
+      expect(SearchgovDomain.exists?(searchgov_domain.id)).to be false
+    end
+
+    context "when the searchgov_domain has associated searchgov_url records" do
+      let!(:searchgov_url1) { searchgov_domain.searchgov_urls.create(url: "https://www.archive.gov/info", hashed_url: "hash1") }
+      let!(:searchgov_url2) { searchgov_domain.searchgov_urls.create(url: "https://www.archive.gov/hmmm", hashed_url: "hash2") }
+
+      it "destroys all associated searchgov_urls" do
+        perform
+        expect(SearchgovUrl.exists?(searchgov_url1.id)).to be false
+        expect(SearchgovUrl.exists?(searchgov_url2.id)).to be false
+      end
+    end
+
+    context "with an acceptable number of URLs" do
+      before do
+        create_batch_of_urls(10_000)
+      end
+
+      it "handles large domains without timing out", :aggregate_failures do
```

Comment:
I do agree that we need to test this but my main concern is that it will drastically increase our testing time. Also do we know if a domain with 10,000 urls is consider a large domain?

I've usually done this kind of performance test separately. We could load a db dump of the urls instead of creating one by one. My succession is to leave this testing and optimization as part of another task and just focus on the deleting feature for now. It seems to me that it requires its own chunk of work.